projects/components/src/action-menu/action-menu.component.ts

```diff
@@ -96,6 +111,38 @@ export class ActionMenuComponent<R, T> {
      */
     @Input() nestedDropdownPosition = 'right-top';
 
+    private _isActionSearchProvider = false;
+    /**
+     * This component is NOT registered by default with with the {@link SpotlightSearchService} to register its actions to show up on the
+     * {@link SpotlightSearchComponent}. However, in the VCD app, some components like the entity details container want to register their
+     * actions with the quick search feature. This switch is used for initializing the {@link #actionSearchProvider} and registering it
+     */
+    @Input() set isActionSearchProvider(val: boolean) {
+        this._isActionSearchProvider = val;
+        if (this.isActionSearchProvider) {
+            this.actionSearchProvider = this.injector.get(ActionSearchProvider);
+            this.updateActionSearchProvider();
+            this.actionSearchProvider.register();
+        }
+    }
+    get isActionSearchProvider(): boolean {
+        return this._isActionSearchProvider;
+    }
+
+    private _actionProviderName: string;
+    /**
+     * Is displayed on the quick search modal as Actions: <actionProviderName>
+     */
+    @Input() set actionProviderName(val: string) {
```

Comment:
In general - if 2 properties go together better group them.
In this specific case it is a bit tedious to know all those interdependencies that in order to provide name, you have to set also `isActionSearchProvider`.